launcher.py

```diff
@@ -222,58 +246,85 @@ def __init__(self):
+        self.src_widgets = [(QCheckBox(f"{src}"), QLineEdit()) for src in up_source]
+        for checkbox, args in self.src_widgets:
+            args.setPlaceholderText("CLI args")
+        self.src_state = [[False, ""] for _ in sources]
+
         
         #Layout for source checkboxes
         self.src_layout = QGridLayout()
         row = 0
         col = 0
-        for checkbox in self.src_checkboxes:
+        for checkbox, args in self.src_widgets:
             self.src_layout.addWidget(checkbox, row, col)
+            self.src_layout.addWidget(args, row + 1, col)
             col += 1
             if col == 3:  # Three checkboxes per row
                 col = 0
-                row += 1
+                row += 2
 
         source_list = QWidget()
         source_list.setLayout(self.src_layout)
         source_list.setContentsMargins(0, 0, 0, 0)
 
+
         #Connect checkbox state to signals to detect which sources were selected 
-        for checkbox in self.src_checkboxes:
-            checkbox.stateChanged.connect(self.update_selected)
+        for i, (checkbox, args) in enumerate(self.src_widgets):
+            # https://docs.python.org/3/faq/programming.html#why-do-lambdas-defined-in-a-loop-with-different-values-all-return-the-same-result
+            def stateChanged(state, i=i):
+                self.src_state[i][0] = True if state == 2 else False
+            def textChanged(text, i=i):
+                # print(f"text change {i}")
+                self.src_state[i][1] = text
+            checkbox.stateChanged.connect(stateChanged)
+            args.textChanged.connect(textChanged)
 
         #Create a sink label
         sink = QLabel(self)
         sink.setText("Sinks:")
-        sink.setGeometry(20, 200, 50, 10)
+        sink.setGeometry(20, 230, 50, 10)
         sink.setContentsMargins(0,0,0,0)
 
         #Create checkboxes for each sink option 
         sinks = self.modules.get("sinks")
         up_sink = [sink.capitalize() for sink in sinks]
         self.sink_dict = {sink: i + 1 for i, sink in enumerate(up_sink)}
-        self.sink_checkboxes = [QCheckBox(f"{sink}") for sink in up_sink]
+        self.sinks_widgets = [(QCheckBox(f"{sink}"), QCheckBox(f"stdout?")) for sink in up_sink]
         
         #Layout for sink checkboxes
         self.sink_layout=QGridLayout()
         row = 0
         col = 0
-        for checkbox in self.sink_checkboxes:
-            self.sink_layout.addWidget(checkbox, row, col)
+        for checkbox, std_out in self.sinks_widgets:
+            group_widget = QWidget()
+            group_layout = QVBoxLayout()
+            group_layout.addWidget(checkbox)
+            group_layout.addWidget(std_out)
+            group_widget.setLayout(group_layout)
+            group_widget.setStyleSheet("border: 1px solid black; padding: 5px;")
+            self.sink_layout.addWidget(group_widget, row, col)
             col += 1
-            if col == 3:
+            if col == 5:  # Three groups per row
```

Comment:
I don't really like how this is hardcoded but it's probably fine (too much work to do it cleanly)